python/lsst/meas/astrom/catalogStarSelector.py

```diff
@@ -76,25 +66,72 @@ def __call__(self, source):
             return False
         return True
 
-class CatalogStarSelector(object):
-    ConfigClass = CatalogStarSelectorConfig
-    usesMatches = True # selectStars uses (requires) its matches argument
+## \addtogroup LSST_task_documentation
+## \{
+## \page CatalogStarSelectorTask
+## \ref CatalogStarSelectorTask_ "CatalogStarSelectorTask"
+## \copybrief CatalogStarSelectorTask
+## \}
 
-    def __init__(self, config=None):
-        """Construct a star selector that uses second moments
+class CatalogStarSelectorTask(object):
+    """!Select stars based on a reference catalog
 
-        This is a naive algorithm and should be used with caution.
+    @anchor CatalogStarSelectorTask_
+    
+    @section meas_astrom_catalogStarSelector_Contents  Contents
 
-        @param[in] config: An instance of CatalogStarSelectorConfig
-        """
-        if not config:
-            config = CatalogStarSelector.ConfigClass()
+     - @ref meas_astrom_catalogStarSelector_Purpose
+     - @ref meas_astrom_catalogStarSelector_Initialize
+     - @ref meas_astrom_catalogStarSelector_IO
+     - @ref meas_astrom_catalogStarSelector_Config
+     - @ref meas_astrom_catalogStarSelector_Debug
+
+    @section meas_astrom_catalogStarSelector_Purpose  Description
+
+    Select stars using a match list: select sources where the matching reference object is unresolved,
+    plus the source passes the following tests:
+    - no flag from config.badFlags is set
+    - psf flux >= config.fluxLim
+    - psf flux <= config.fluxMax (not checked if fluxMax == 0)
+
+    @section meas_astrom_catalogStarSelector_Initialize  Task initialisation
+
+    @copydoc \_\_init\_\_
+
+    @section meas_astrom_catalogStarSelector_IO  Invoking the Task
+
+    Like all star selectors, the main method is `run`.
```

Comment:
Should call out the fact that this is an unusual `StarSelector` for which `usesMatches=True`, and hence that argument is required.